### ISSUE TITLE
[Feature Request]: Porter

### ISSUE THREAD

**Issue by ross-p-smith** [devcontainers-contrib/features#427]
### Requested Feature

Porter is a bundle manager using containers: - https://porter.sh/install/

It would be great that this was a dev container feature. Below is my script that is used to install from a dockerfile.

```bash
#!/bin/bash
set -o errexit
set -o pipefail
set -o nounset
# Uncomment this line to see each command for debugging (careful: this will show secrets!)
# set -o xtrace

export PORTER_HOME=${PORTER_HOME:-~/.porter}
export PORTER_MIRROR=${PORTER_MIRROR:-https://cdn.porter.sh}
PORTER_VERSION=${PORTER_VERSION:-latest}

echo "Installing porter@$PORTER_VERSION to $PORTER_HOME from $PORTER_MIRROR"

mkdir -p "$PORTER_HOME/runtimes"

curl -fsSLo "$PORTER_HOME/porter" "$PORTER_MIRROR/$PORTER_VERSION/porter-linux-amd64"
chmod +x "$PORTER_HOME/porter"
ln -s "$PORTER_HOME/porter" "$PORTER_HOME/runtimes/porter-runtime"
echo "Installed $("${PORTER_HOME}"/porter version)"

"${PORTER_HOME}/porter" mixin install exec --version "$PORTER_VERSION"
"${PORTER_HOME}/porter" mixin install terraform --version "$PORTER_TERRAFORM_MIXIN_VERSION"
"${PORTER_HOME}/porter" mixin install az --version "$PORTER_AZ_MIXIN_VERSION"

"${PORTER_HOME}/porter" plugin install azure --version "$PORTER_AZURE_PLUGIN_VERSION"

chown -R "${USERNAME}" "${PORTER_HOME}"

echo "Installation complete."
```

**Comment by danielbraun89** [comment]
@ross-p-smith  Thanks for suggesting this feature! I have implemented, but with a few changes, that derived from the design guidelines of this repo. 
The changes are:

1. Instead of running the actuall script, I am calling the official script of https://porter.sh/ , this will make the feature more maintainable.
2. Instead of using PORTER_HOME of `~/porter` , I am installing to `/usr/local/porter` instead.  This allows me to have a fixed path and to include it into the user `PATH` thru the usage of devcontainer's `containerEnv` field.  (This is a common pattern as can be seen in the official feature repo https://github.com/search?q=repo%3Adevcontainers%2Ffeatures%20containerEnv&type=code)
3. Instead of installing only az , terraform mixins, I opted to install them all because a. I cant determine which are more important b. Having a enable boolean flag for each of them will bloat the options num c. There seem to be a finit low number of them, so no huge impact on the system. Same goes for plugins. Also have added tests for each mixin and plugin

PR:  #432
